packages/client/tests/functional/metrics/tests.ts

```diff
@@ -0,0 +1,101 @@
+import testMatrix from './_matrix'
+
+// @ts-ignore this is just for type checks
+declare let prisma: import('@prisma/client').PrismaClient
+// @ts-ignore this is just for type checks
+declare let PrismaClient: typeof import('@prisma/client').PrismaClient
+
+testMatrix.setupTestSuite(() => {
+  beforeAll(async () => {
+    const email = 'user@example.com'
+    await prisma.user.create({
+      data: {
+        email,
+      },
+    })
+
+    await prisma.user.findFirst({ where: { email } })
+  })
+
+  test('returns metrics in prometheus format', async () => {
+    const metrics = await prisma.$metrics.prometheus({ globalLabels: { foo: 'bar' } })
+
+    expect(metrics).toContain('query_total_operations')
+    expect(metrics).toContain('query_total_queries')
+    expect(metrics).toContain('pool_active_connections')
+    expect(metrics).toContain('pool_active_connections')
+    expect(metrics).toContain('pool_idle_connections')
+    expect(metrics).toContain('pool_wait_count')
+    expect(metrics).toContain('query_active_transactions')
+    expect(metrics).toContain('pool_wait_duration_ms_bucket')
+    expect(metrics).toContain('query_total_elapsed_time_ms_bucket')
+  })
+
+  test('includes global labels in prometheus format', async () => {
+    const metrics = await prisma.$metrics.prometheus({ globalLabels: { label1: 'value1', label2: 'value2' } })
+
+    expect(metrics).toContain('{label1="value1",label2="value2"}')
+  })
+
+  test('returns metrics in json format', async () => {
+    const { counters, gauges, histograms } = await prisma.$metrics.json()
```

Comment:
So, here, rather than doing snaphots and introducing dependency on engine implementation details, we assert that returned metrics has the data and general properties of the data kind of make sense. This should make test more resilient to engine internal changes, which will cause the changes to the metrics.